### PR TITLE
Add questions page

### PR DESCRIPTION
**Pull request by Maianki**
## Added questions page and rules card

- [ ] Rules card
- [ ] questions page

### Thanks for reviewing my PR

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
         <Route path='/' element={<Home />}></Route>
         <Route path='/login' element={<>Login</>}></Route>
-        <Route path='/rules' element={<QuestionCard />}></Route>
+        <Route path='/rules' element={<RulesCard />}>
+          <Route path=':quizId' element={<RulesCard />}></Route>
+        </Route>
+        <Route
+          path='/questions/:quizId/:questionNumber'
+          element={<Questions />}
+        ></Route>
```
Comment (anik31) [inline src/App.js:20]: appreciation - good that you're having different route for different question

File: src/components/Card/QuestionCard/Answer/Answer.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { useGlobal } from "context/global-context";
+
+export default function Answer({ option, answer }) {
+  const { globalDisptacher } = useGlobal();
+  //   const correctAnswerClass = answer === option ? `correct-answer` : ``;
```
Comment (sainath7878) [inline src/components/Card/QuestionCard/Answer/Answer.js]: remove commented code

File: src/components/Card/QuestionCard/Answer/Answer.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  //   const inCorrectAnswerClass = answer !== option ? `incorrect-answer` : ``;
```
Comment (anik31) [inline src/components/Card/QuestionCard/Answer/Answer.js]: suggestion - avoid pushing commented code

File: src/components/Card/QuestionCard/Answer/Answer.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  const handleAnswer = (option) => {
+    globalDisptacher({ type: "SET_ANSWERS", payload: option });
+  };
+  return (
+    <li
+      className={`list-unstyled question-option text-center `}
+      role='button'
```
Comment (anik31) [inline src/components/Card/QuestionCard/Answer/Answer.js:12]: appreciation - good that you're using the role attribute, it's good for accessibility

File: src/components/Card/QuestionCard/Answer/Answer.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+export default function Answer({ option, answer }) {
+  const { globalDisptacher } = useGlobal();
+  //   const correctAnswerClass = answer === option ? `correct-answer` : ``;
+  //   const inCorrectAnswerClass = answer !== option ? `incorrect-answer` : ``;
+  const handleAnswer = (option) => {
+    globalDisptacher({ type: "SET_ANSWERS", payload: option });
+  };
+  return (
+    <li
+      className={`list-unstyled question-option text-center `}
```
Comment (anik31) [inline src/components/Card/QuestionCard/Answer/Answer.js:11]: suggestion - you can write class names inside inverted commas, I don't think curly braces & template literals are required here.

File: src/components/Modal/Modal.jsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import "./modal.css";
+
+export function Modal({ children }) {
+  const { handleModal } = useGlobal();
+  return createPortal(
+    <div>
+      <div className='overlay' onClick={() => handleModal(false)}></div>
+      <span></span>
+      <div className={"modal-body"}>{children}</div>
+    </div>,
```
Comment (anik31) [inline src/components/Modal/Modal.jsx:13]: appreciation - good that you're using portal for modal

**Review by sainath7878 — APPROVED** [review]
Code is clean and understandable. Awesome UI too. Looks good to merge!

**Review by anik31 — APPROVED** [review]
Nice work @Maianki , everything is working fine in the preview

File: src/components/Card/QuestionCard/Answer/Answer.js (hunk 5 above)
Comment (Maianki) [reply src/components/Card/QuestionCard/Answer/Answer.js:11]: Yeah, sure I will change it.

File: src/components/Card/QuestionCard/Answer/Answer.js (hunk 3 above)
Comment (Maianki) [reply src/components/Card/QuestionCard/Answer/Answer.js]: sure, will remove it.

File: src/components/Card/QuestionCard/Answer/Answer.js (hunk 2 above)
Comment (Maianki) [reply src/components/Card/QuestionCard/Answer/Answer.js]: sure, will remove it

**Comment by Maianki** [comment]
Thanks for reviewing @anik31 @sainath7878 . Done the said changes. Merging PR.

**Pull request merged**